src/page.rs

```diff
@@ -54,16 +54,20 @@ impl Page {
         let mut urls: Vec<String> = Vec::new();
         let selector = Selector::parse("a").unwrap();
 
-        for element in self.get_html().select(&selector) {
-            match element.value().attr("href") {
-                Some(href) => {
+        let html = self.get_html();
+        let anchors = html.select(&selector)
+                          .filter(|a| a.value().attrs().any(|attr| attr.0 == "href"));
 
-                    // Keep only links for this domains
-                    match href.find('/') {
-                        Some(0) => urls.push(format!("{}{}", domain, href)),
-                        Some(_) => (),
-                        None => (),
-                    };
+        for anchor in anchors {
+            match anchor.value().attr("href") {
+                Some(href) => {
+                    if href.starts_with(domain) {
+                        urls.push(format!("{}", href));
+                    } else if href.starts_with('/') {
+                        urls.push(format!("{}{}", domain, href))
+                    } else if !href.starts_with("http") && !href.starts_with("/") {
+                        urls.push(format!("{}/{}", domain, href))
```

Comment:
This case is more complicated because of [relative file path ](https://www.w3schools.com/html/html_filepaths.asp). For example, theses case may not works

- `./test.html`
- `../../test.html`

So In this special case, this would be

```rs
// this is pseudocode
let absolutePath = getAbsolutePathFromPage(href, pageUrl)
urls.push(format!("{}/{}", domain, absolutePath))
```

If you decide to try to implement this, can you also add a unit test?

Reply:
I just added the needed method and used the Url crate to better handle URLs. Please let me know what you think and correct any bad or poor code.

Reply:
It seems good. :+1: